### ISSUE TITLE
Add Websocket support

### ISSUE THREAD

**Issue by wsargent** [playframework/play-ws#79]
AsyncHTTPClient has underlying websocket support -- we should be able to do this with Akka Streams fairly easily.

**Comment by ktoso** [comment]
Yeah, websockets are rather simple with Akka, pretty low level API though so I think if you could package it up in WS in a nice API that'd be awesome actually

**Comment by wsargent** [comment]
We can do the same thing we do on the client side: 

```
import play.api.mvc._
import play.api.libs.streams._

class Controller1 @Inject() (implicit system: ActorSystem, materializer: Materializer) {
  def socket = WebSocket.accept[String, String] { request =>
    ActorFlow.actorRef(out => MyWebSocketActor.props(out))
  }
}
```

https://www.playframework.com/documentation/2.5.x/ScalaWebSockets

**Comment by wsargent** [comment]
Also see https://github.com/eed3si9n/gigahorse/blob/0.3.x/asynchttpclient/src/main/scala/gigahorse/support/asynchttpclient/WebSocketListener.scala
